### PR TITLE
APIMAN-1339: Support self signed certificates from keycloak in vertx

### PR DESCRIPTION
**Pull request by volkflo**
This is a fix for https://issues.jboss.org/browse/APIMAN-1339

Without this fix it is not possible to connect vertx with a standalone keycloak installation that is secured by a self signed certificate.

Please let me know what you think :)

### CODE REVIEW

File: gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/auth/KeycloakOAuthFactory.java [hunk 1]
```diff
@@ -143,8 +148,42 @@ public AuthHandler addAuthorities(Set<String> authorities) {
...
+        if (config.containsKey("allow-any-hostname")) {
+            Object value = config.getValue("allow-any-hostname");
+            if (value.equals(true) || value.equals("true")) {
+                options.setVerifyHost(false);
+            }
+        }
+
+        if (config.containsKey("disable-trust-manager")) {
+            Object value = config.getValue("disable-trust-manager");
+            if (value.equals(true) || value.equals("true")) {
```
Comment (volkflo) [inline gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/auth/KeycloakOAuthFactory.java]: With `if (value.equals(true) || value.equals("true")) ` you are able to use System Properties in the conf-X.json like so:

```
"disable-trust-manager": "${devMode}",
"allow-any-hostname" : "${allowAnyHost}"
```

**Comment by EricWittmann** [comment]
@msavy any problems with this?

**Comment by volkflo** [comment]
@EricWittmann What is the status of this? Would be very nice if we could include this :)

**Comment by msavy** [comment]
I'm just slightly concerned this might break things for something other than Keycloak, but otherwise I'm happy for this to be merged

**Comment by EricWittmann** [comment]
Rolling the dice!  :)

**Comment by volkflo** [comment]
Thanks for merging, if something goes wrong with providers other than keycloak, you can blame me for this :)

**Pull request merged**